src/main/java/ai/rev/speechtotext/CustomVocabulariesClient.java

```diff
@@ -0,0 +1,141 @@
+package ai.rev.speechtotext;
+
+import ai.rev.speechtotext.helpers.ClientHelper;
+import ai.rev.speechtotext.models.CustomVocabulary;
+import ai.rev.speechtotext.models.CustomVocabularyInformation;
+import ai.rev.speechtotext.models.asynchronous.RevAiJobOptions;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * The CustomVocabularyClient object provides methods to submit custom vocabularies and retrieve
+ * related job information.
+ */
+public class CustomVocabulariesClient {
+
+  private OkHttpClient client;
+
+  /** Interface that CustomVocabularyClient methods use to make requests */
+  public ApiInterface apiInterface;
+
+  /**
+   * Constructs the custom vocabulary client used to create custom vocabularies for streaming. The
+   * user access token can be generated on the website at <a
+   * href="https://www.rev.ai/access_token">https://www.rev.ai/access_token</a>.
+   *
+   * @param accessToken Rev.ai authorization token associate with the account.
+   */
+  public CustomVocabulariesClient(String accessToken) {
+    if (accessToken == null) {
+      throw new IllegalArgumentException("Access token must be provided");
+    }
+    this.client = ClientHelper.createOkHttpClient(accessToken);
+    Retrofit retrofit = ClientHelper.createRetrofitInstance(client);
+    this.apiInterface = retrofit.create(ApiInterface.class);
+  }
+
+  /**
+   * This methods makes a POST request to the /vocabularies endpoint, starts an asynchronous job to
+   * process the vocabularies and returns a {@link CustomVocabularyInformation} object.
+   *
+   * @param metadata Optional information that can be provided.
+   * @param customVocabularies List of {@link CustomVocabulary} objects.
+   * @param callbackUrl The callback url that Rev.ai will send a POST to when the job has finished.
+   * @return A {@link CustomVocabularyInformation} object.
+   * @throws IOException If the response has a status code > 399.
+   * @throws IllegalArgumentException If the list of custom vocabularies is null.
+   * @see CustomVocabularyInformation
+   */
+  public CustomVocabularyInformation submitCustomVocabularies(
+      String metadata, List<CustomVocabulary> customVocabularies, String callbackUrl)
```

Comment:
Are these parameters formatted correctly? It seems strange to line break before the first parameter in order to line wrap.

Reply:
The other question I have is about the ordering of the parameters. I would normally order them by importance with the optional parameters coming last: `customVocabularies`, `metadata`, `callbackUrl`.

I think that you have chosen this order so that you can override the method so that both metadata and callbackurl can be optional. I think it is only because the types of theparameters are List<CustomVocabulary>, string, string. The last two parameters would be indistinguishable if you did not have the order that you have written. Is this correct?

I worry that in the effort to create an exhaustive list of overrides we have unintentionally made the ordering of the parameters too confusing. If necessary, I would prefer to have a non-exhastive list of overrides and a more clear parameter ordering. With just a single override method, extensive customization of the parameters can be done by configuring a CustomVocabularyJobOption object with the desired properties and injecting it as a parameter to the function.

Reply:
Your reasoning as to the ordering is correct. I too was feeling that we should have an options object but in order to follow the other SDK implementations it requires this parameter ordering. Are we ok with deviating here?